pyccel/ast/numpyext.py

```diff
@@ -425,7 +425,7 @@ class NumpyImag(PythonImag):
     """
     def __new__(cls, arg):
         if not isinstance(arg.dtype, NativeComplex):
-            return arg
+            return NumpyZeros(arg.shape, dtype='int')
```

Comment:
Right class :) . Are you sure about the dtype?

Reply:
(e.g. `NativeReal`)

Reply:
ah sorry 😐 , that was just for some tests.